File: Arrays/Leetcode/Easy/TwoSum/TwoSum.java
// https://leetcode.com/problems/two-sum/
import java.util.HashMap;
import java.util.Map;

public class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> mp = new HashMap<>();

        for(int i=0; i<nums.length; i++) {
            int t = mp.getOrDefault(target - nums[i], -1);
            if(t != -1) {
                return new int[]{i, t};
            } else {
                mp.put(nums[i], i);
            }
        }
        return new int[]{};
    }
}

File: Arrays/Leetcode/Easy/TwoSum/TwoSum.cpp
// https://leetcode.com/problems/two-sum/
#include<bits/stdc++.h>
using namespace std;
vector<int> twoSum(vector<int> &nums, int target)
{
    unordered_map<int, int> mp;
    mp[nums[0]] = -1;
    for (int i = 1; i < nums.size(); i++)
    {
        int t = mp[target - nums[i]];
        if (t)
        {
            if (t == -1)
            {
                return {0, i};
            }
            else
            {
                return {i, t};
            }
        }
        else
        {
            mp[nums[i]] = i;
        }
    }
    return {0, 0};
}